packages/node_modules/@webex/plugin-meetings/sample/app.js

```diff
@@ -340,107 +498,113 @@ document.getElementById('stop-receiving-video').addEventListener('click', () =>
-      .withMedia()
-      .build()
-      .getMedia()
-      .withId()
-      .withStream(meeting.mediaProperties.mediaPeerConnection)
-      .build()
-      .getStream()
-      .on('data', (raw) => {
-        console.log(raw);
-      });
-  }
-});
+    const audioSenderId = '0';
+    const audioReceiverId = '1';
+    const videoSenderId = '2';
+    const videoReceiverId = '3';
+    const shareSenderId = '4';
+    const shareReceiverId = '5';
+
+    const options = {
+      useConfig: true,
+      senders: [
+        {
+          id: audioSenderId,
+          correlate: 'audio', // NECESSARY KEY
+          history: true,
+          aggregator: true,
+          interval: true,
+          onData: (data) => {
+            console.log('audioSender->', data);
+          },
+          onClose: (error) => {
+            console.warn('audioSender->', error);
+          },
+          onEvent: (event) => {
+            console.error('audioSender->', event);
+          }
+        }
+      ],
+      receivers: [
+        {
+          id: audioReceiverId,
+          correlate: 'audio', // NECESSARY KEY
+          onData: (data) => { console.log('audioReceiver->', data); },
+          onClose: (error) => { console.warn('audioReceiver->', error); },
+          onEvent: (event) => { console.error('audioReceiver->', event); }
+        }
+      ]
+    };
 
-document.getElementById('getStatsFilter').addEventListener('click', () => {
-  if (meeting) {
-    // meeting
-    //   .getStats()
-    //   .withMedia()
-    //   .build()
-    //   .getMedia()
-    //   .withId()
-    //   .withEvents()
-    //   .withHistory()
-    //   .withAggregator()
-    //   .withInterval()
-    //   .withFilter(meeting.mediaProperties.mediaPeerConnection)
-    //   .build()
-    //   .onData((filtered) => {
-    //     console.log(filtered);
-    //   });
-    // OR
-    // const stats = meeting.getStats().withId().withMedia().build();
-    // const media = stats.getMedia()...
-    // media.doData((filtered) => {
-    //  do stuff
-    // })
-    // OR
-    meeting.getStats({
-      media: {
-        id: true,
-        history: true,
-        interval: true,
-        aggregator: true,
-        filter: meeting.mediaProperties.mediaPeerConnection,
-        events: true
-      }
-    });
-    meeting
-      .getStats()
-      .getMedia()
-      .onData((filtered) => {
-        console.log(filtered);
-      });
+    meeting.getStats(options, true);
+
+    // notice no meeting is needed, and no correlate key is required, this is the manual builder method
     meeting
       .getStats()
-      .getMedia()
-      .getEvents()
-      .on('stats:update', (stats) => {
-        console.log(stats);
-      });
-  }
-});
-
-document.getElementById('getAggregation').addEventListener('click', () => {
-  if (meeting) {
-    console.log(
-      meeting
-        .getStats()
-        .getMedia()
-        .getAggregator()
-        .get()
+      .addSenderStats(shareSenderId)
+      .withHistory()
```

Comment:
Would just `meeting.withHistory()` work? If so, does it make sense to rename `addHistory` to be consistent with `addSenderStats`?

Reply:
meeting.withHistory would not make sense to me, there are multiple transceivers collecting data, (6 with our webrtc implementation)
it should be attached to the stats sender object that it's associated to.

Reply:
Okay - it sounds like anything starting with `with` is not meant to attach tot he main object, but rather it augments stats after the fact. What's the best way to communicate this to future contributors? A note about how to use it this way? A config object passed to `addSenderStats` like `addSenderStats(shareSenderId, {history: true})`? I'm trying to think of how to make it clear months from now so someone doesn't try `meeting.withHistory()`. Does that make sense? Am I overthinking it?

Reply:
withHistory does not augment the meeting object. the structure is as follows

```
meeting = {
...
stats = { // can be instantiated or got from getStats() call
senders = { // added to using addSenderStats
senderStat1 = { // can attach anything you want to the senderStat
history, interval, filter, etc
},
...
senderStatN
}
}
}

so when you say meeting.getStats() you are getting the stats object, when you then say addSenderStats() you are getting back the sender object, and then you are chaining an attached history module to that senderStat object. it's just chaining return values to make a single call. 

this would be the equivalent
const stats = meeting.getStats()
const senderStats = stats.addSenderStats()
const augmentedSenderStats = senderStats.withHistory()

if someone calls meeting.withHistory it will say withHistory is undefined.  IMO it's clear that stats is a separate container living on the meeting object, and that stats object has actions to take